### PR TITLE
Bump golangci-lint

### LINKED ISSUE #609: Bump GOLANGCI-LINT
Bump golang-ci lint 
 - bump GH action
 - consider to extend with new linters
 - Fix issues if occur

### PR DESCRIPTION
**Pull request by kuritka**
closes #609
Golangci-lint has been greatly expanded with new linter types, some of which I have integrated.

With the new linters came also changes in the code, mainly
 - introduction of constants for repeating text
 - throwing out deprecated code

Some of the linters were not suitable and have been commented out
 - golint is deprecated, revive is instead
 - whitespace - it's bad to have empty lines in the code, in my opinion it degrades clarity
 - funlen - watches the length of functions and their names
 - exhaustive - completing missing constants in the switch (not only default)
 - gomnd - force to have any number in constant. E.g. mark if len(x) > 2 ,
 - gofmt - is part of goimports

I had to change linter configuration for these:
 - gochecknoinits - it is bothered by wrong use of init function - init() funcs in generated code
 - dupl - duplicates in the code (this only applies to testing, where duplicates happen frequently. We don't mind it there)

In addition I updated build pipeline.

Signed-off-by: kuritka <kuritka@gmail.com>

### CODE REVIEW

**Comment by kuritka** [comment]
@here: will require upgrade golangci locally:  https://golangci-lint.run/usage/install/#macos

**Review by jkremser — APPROVED** [review]

File: .golangci.yaml [hunk 1]
```diff
@@ -7,18 +7,55 @@ linters:
...
+    - exportloopref
+    - goconst
+    - goprintffuncname
+    - nolintlint
+    - dupl
+    - gochecknoinits
+
+ # don't enable:
+ #  - golint        # deprecated
+ #  - whitespace
```
Comment (ytsarev) [inline .golangci.yaml:39]: why not? :)

File: controllers/providers/dns/factory_test.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ import (
 
 func TestFactoryInfoblox(t *testing.T) {
 	// arrange
-	client := fake.NewFakeClientWithScheme(scheme.Scheme, []runtime.Object{}...)
+	client := fake.NewClientBuilder().WithScheme(scheme.Scheme).WithRuntimeObjects([]runtime.Object{}...).Build()
```
Comment (ytsarev) [inline controllers/providers/dns/factory_test.go:36]: Is it dictated by newer fake client version?

**Review by ytsarev — COMMENTED** [review]
Left couple of comments otherwise looks good

File: .golangci.yaml (hunk 1 above)
Comment (kuritka) [reply .golangci.yaml:39]: I had problem with empty lines we keep in the code, which delimits block of code, e.g.: we will remove empty lines here: [this](https://github.com/k8gb-io/k8gb/blob/master/controllers/gslb_controller.go#L128-L184)

File: controllers/providers/dns/factory_test.go (hunk 2 above)
Comment (kuritka) [reply controllers/providers/dns/factory_test.go:36]: It's deprecated. see upstrean function in controller runtime:  [uses NewClientBuilder anyway and  comment says it's deprecated](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/fake/client.go#L73-L80)

**Review by somaritane — APPROVED** [review]
lgtm

**Review by k0da — APPROVED** [review]

**Review by ytsarev — APPROVED** [review]

**Pull request merged**